views/pages/AccountPage.jsx

```diff
@@ -33,22 +29,18 @@ const AccountPage = () => {
 
       reader.readAsDataURL(files[0])
       reader.onload = (e) => {
-        setUser((prevState) => {
-          return {
-            ...prevState,
-            icon: e.target.result,
-          }
-        })
+        setFormValues({ ...formValues, icon: e.target.result })
       }
     }
   }
 
-  const handleSubmit = async (e, formValues) => {
+  const handleSubmit = async (e) => {
     try {
       e.preventDefault()
 
-      await api.users.update(user.uid, formValues)
+      const updatedUser = await api.users.update(user.uid, formValues)
 
+      setUser(updatedUser)
```

Comment:
Unsure what happened, but now the profile image isn't updating on the sidebar unless it's refreshed
```suggestion
      setUser(formValues)
```

Reply:
I don't think we want to set the user to the form values. When we save the user and get a returned user from the server we should replace the user we have in our context with that updated user.